### DISCUSSION TITLE
How to handle validation errors thrown from Remix actions?

### DISCUSSION THREAD

**Discussion by alexluong** [esamattis/react-zorm#8] [General]
Hi, really like the idea of `react-zorm` and am curious if you have any thoughts on this topic?

Let's say for some reason the form go through even if there's some validation error, how should we handle the thrown response from Remix action?

**Comment by esamattis** [comment]
@alexluong  Still learning Remix myself too but I'd say just let the Remix error boundary handle it since it should not really happen. But if you really need more fine grained handling, for example if you really don't want to lose form state in any situation, you should use `safeParseForm()` instead of `parseForm()` in the action handler and return some error flag and handle it with `useActionData()` on your form page.
